kimchi/src/circuits/gate.rs

```diff
@@ -225,35 +238,47 @@ impl<F: FftField> LookupInfo<F> {
 
     /// Each entry in `kinds` has a corresponding selector polynomial that controls whether that
     /// lookup kind should be enforced at a given row. This computes those selector polynomials.
-    pub fn selector_polynomials(
+    #[allow(clippy::type_complexity)]
+    pub fn selector_polynomials_and_tables(
         &self,
         domain: &EvaluationDomains<F>,
         gates: &[CircuitGate<F>],
-    ) -> Vec<E<F, D<F>>> {
+    ) -> (Vec<E<F, D<F>>>, Vec<Vec<Vec<F>>>) {
```

Comment:
If I understand correctly, this vec of vec of vec is simply a list of all tables before being combined. We should have a type for a table (and a type for a combined table)